File: javabackend/src/main/java/mft/dev/javabackend/service/UtenteService.java
package mft.dev.javabackend.service;

import mft.dev.javabackend.dto.insert.request.UtenteInsertRequestDTO;
import mft.dev.javabackend.dto.update.UtenteUpdateRequestDTO;
import mft.dev.javabackend.model.Utente;

import java.util.List;

public interface UtenteService {

    List<Utente> getUtenti();

    Utente getUtente(long id);

    Utente insertUtente(UtenteInsertRequestDTO dto);

    Utente updateUtente(long id, UtenteUpdateRequestDTO dto);

    void deleteUtente(long id);
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/service/UtenteService.kt
package mft.dev.kotlinbackend.service

import mft.dev.kotlinbackend.dto.insert.UtenteInsertRequestDTO
import mft.dev.kotlinbackend.dto.update.UtenteUpdateRequestDTO
import mft.dev.kotlinbackend.model.Utente

interface UtenteService {
    fun getUtenti(): MutableList<Utente>

    fun getUtente(id: Long): Utente

    fun insertUtente(dto: UtenteInsertRequestDTO): Utente

    fun updateUtente(id: Long, dto: UtenteUpdateRequestDTO): Utente

    fun deleteUtente(id: Long)
}